ci/envs/311-dev.yaml

```diff
@@ -25,6 +24,7 @@ dependencies:
     # dev versions of packages
     - --pre --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --extra-index-url https://pypi.fury.io/arrow-nightlies/ --extra-index-url https://pypi.org/simple
     - fiona
+    - numpy>=2.0.0.dev
```

Comment:
We can't use a pin here, we would have to update it with every numpy release.

Reply:
Yeah it's not ideal, but pip doesn't seem to resolve to the new version without it - https://github.com/geopandas/geopandas/actions/runs/6516980398/job/17701050086 (or [this is the raw log](https://pipelinesghubeus23.actions.githubusercontent.com/PKwT4dmQuTdEdCCMou00pYMdbWiNGGYwovIYD0ZkwVHPZlJPHz/_apis/pipelines/1/runs/4702/signedlogcontent/16?urlExpires=2023-10-15T07%3A41%3A29.0074736Z&urlSigningMethod=HMACV1&urlSignature=Pg5rdMeX5ee1DZyM4kMhWku7MN5xUiDo2f%2BxpiFIDkw%3D). Having a look at the difference, asking for new numpy gives us an older version of contourpy (matplotlib dependency), which doesn't pin numpy <2.0 https://github.com/contourpy/contourpy/pull/268.

I don't think we would need to update this with every release, it's just for the specific instance where a less than pin is introduced. (Technically this means we are getting a version of contourpy before the pin so this may also cause issues, but it seems like it was done out of caution based on the recommendation, rather than a known issue)

(We may also be able to install contourpy via git, but the numpy<2 pin is still present on main for them, so this would also resolve to numpy <2)